src/Service/CheckoutService.php

```diff
@@ -34,25 +33,10 @@ class CheckoutService extends Checkout
      */
     private $client;
 
-    /**
-     * @var LoggerInterface
-     */
-    private $logger;
-
     public function __construct(
-        ClientService $client,
-        LoggerInterface $logger
+        Client $client
     ) {
         $this->client = $client;
-        $this->logger = $logger;
-    }
-
-    public function startClient($salesChannelId)
-    {
-        try {
-            parent::__construct($this->client->getClient($salesChannelId));
-        } catch (AdyenException $e) {
-            $this->logger->error($e->getMessage());
-        }
+        parent::__construct($this->client);
     }
```

Comment:
there's no need to declare this service in the DI container since we always needed to call `startClient()` with the current context's salesChannelId, we can simplify this by instantiating it injecting the right client at runtime instead.